makefiles/mcuboot.mk

```diff
@@ -1,16 +1,22 @@
+.PHONY: mcuboot mcuboot-create-key mcuboot-flash-bootloader mcuboot-flash
+
 ifdef MCUBOOT_SLOT0_SIZE
 
-IMGTOOL ?= $(RIOTTOOLS)/mcuboot/imgtool.py
-override IMGTOOL := $(abspath $(IMGTOOL))
+IMGTOOL ?= $(abspath $(RIOTTOOLS)/mcuboot/imgtool.py)
 
-BINFILE ?= $(BINDIR)/$(APPLICATION).bin
 SIGN_BINFILE = $(BINDIR)/signed-$(APPLICATION).bin
 MCUBOOT_KEYFILE ?= $(BINDIR)/key.pem
-MCUBOOT_BIN ?= $(BINDIR)/mcuboot.bin
-MCUBOOT_BIN_URL ?= http://download.riot-os.org/mynewt.mcuboot.bin
-MCUBOOT_BIN_MD5 ?= 0c71a0589bd3709fc2d90f07a0035ce7
 
-export IMAGE_HDR_SIZE ?= 512
+MCUBOOT_ELF ?= $(ELFFILE:.elf=.mcuboot.elf)
+MCUBOOT_BIN ?= $(MCUBOOT_ELF:.elf=.bin)
+
+MCUBOOT_LOADER_BIN ?= $(BINDIR)/mcuboot.bin
+MCUBOOT_LOADER_BIN_URL ?= http://download.riot-os.org/mynewt.mcuboot.bin
+MCUBOOT_LOADER_BIN_MD5 ?= 0c71a0589bd3709fc2d90f07a0035ce7
+
+IMAGE_HDR_SIZE ?= 512
+
+BUILDDEPS += $(MCUBOOT_KEYFILE) $(MCUBOOT_LOADER_BIN)
```

Comment:
@cladmi This thing is not necessary, right? Those two are already dependencies of other rules.

Should I remove it?
```suggestion
```

Reply:
As shown in #10492 , this should not be here.